crates/sargon/src/system/sargon_os/sargon_os_delete_account.rs

```diff
@@ -0,0 +1,144 @@
+            resource_preferences,
+            authorized_depositors,
+        );
+
+        Ok(manifest)
+    }
+}
+
+// ==================
+// Delete Account (Internal)
+// ==================
+impl SargonOS {
+    async fn get_account_resource_preferences(
+        &self,
+        network_id: NetworkID,
+        account_address: AccountAddress,
+    ) -> Result<Vec<ScryptoAccountRemoveResourcePreferenceInput>> {
+        let gateway_client = GatewayClient::new(
+            self.clients.http_client.driver.clone(),
+            network_id,
+        );
+
+        let mut items: Vec<AccountResourcePreference> = Vec::new();
+        let mut more_to_load = true;
+        let mut cursor: Option<String> = None;
+        while more_to_load {
+            let request = AccountPageRequest::new(
+                account_address,
+                cursor.clone(),
+                GATEWAY_PAGE_REQUEST_LIMIT,
+            );
+            let response =
+                gateway_client.account_resource_preferences(request).await?;
+            items.extend(response.items);
+            cursor = response.next_cursor;
+            more_to_load = cursor.is_some();
+        }
+
+        let items = items
+            .into_iter()
+            .map(ScryptoAccountRemoveResourcePreferenceInput::from)
+            .collect();
+        Ok(items)
+    }
+
+    async fn get_account_authorized_depositors(
+        &self,
+        network_id: NetworkID,
+        account_address: AccountAddress,
+    ) -> Result<Vec<ScryptoAccountRemoveAuthorizedDepositorInput>> {
+        let gateway_client = GatewayClient::new(
+            self.clients.http_client.driver.clone(),
+            network_id,
+        );
+
+        let mut items: Vec<AccountAuthorizedDepositor> = Vec::new();
+        let mut more_to_load = true;
+        let mut cursor: Option<String> = None;
+        while more_to_load {
+            let request = AccountPageRequest::new(
+                account_address,
+                cursor.clone(),
+                GATEWAY_PAGE_REQUEST_LIMIT,
+            );
+            let response = gateway_client
+                .account_authorized_depositors(request)
+                .await?;
+            items.extend(response.items);
+            cursor = response.next_cursor;
+            more_to_load = cursor.is_some();
+        }
+
+        let items = items
+            .into_iter()
+            .map(ScryptoAccountRemoveAuthorizedDepositorInput::try_from)
+            .collect::<Result<Vec<ScryptoAccountRemoveAuthorizedDepositorInput>>>()?;
+        Ok(items)
+    }
+}
+
+impl TryFrom<AccountAuthorizedDepositor>
+    for ScryptoAccountRemoveAuthorizedDepositorInput
+{
+    type Error = CommonError;
+    fn try_from(value: AccountAuthorizedDepositor) -> Result<Self> {
+        let resource_or_non_fungible = ResourceOrNonFungible::try_from(value)?;
+        Ok(resource_or_non_fungible.into())
+    }
+}
+
+impl TryFrom<AccountAuthorizedDepositor> for ResourceOrNonFungible {
```

Comment:
This logic would be a lot easier if `AccountAuthorizedDepositor`  would hold a `NonFungibleGlobalId`, rather than a `String`. Wasn't able to make it implement `Serializable/Deserializable`, that's why it turns up implementing `TryFrom<>` inside this conversion.